var/spack/repos/builtin/packages/py-petsc4py/package.py

```diff
@@ -42,7 +43,8 @@ class PyPetsc4py(PythonPackage):
     depends_on('petsc+mpi', when='+mpi')
     depends_on('petsc~mpi', when='~mpi')
     depends_on('petsc@develop', when='@develop')
-    depends_on('petsc@3.14:3.14.99', when='@3.14:3.14.99')
+    depends_on('petsc@3.14.2:3.14.99', when='@3.14.2:3.14.99')
+    depends_on('petsc@3.14:3.14.1', when='@3.14')
```

Comment:
Should this be:
```suggestion
    depends_on('petsc@3.14:3.14.1', when='@3.14.0:3.14.1')
```
I can never remember how shortened versions work.

Reply:
Ah - my mistake. Pushed a fix.